app/routes/surveys/AddSubmissionRoute.js

```diff
@@ -27,9 +43,7 @@ const mapDispatchToProps = { submitFunction: addSubmission };
 
 export default compose(
   replaceUnlessLoggedIn(LoginPage),
-  prepare(({ params: { surveyId } }, dispatch) => dispatch(fetch(surveyId)), [
-    'params.surveyId'
-  ]),
+  prepare(loadData, ['params.surveyId', 'currentUser.id']),
   connect(mapStateToProps, mapDispatchToProps),
   loadingIndicator(['survey.questions', 'survey.event.cover'])
```

Comment:
Maybe add the `fetching` for `Suervey.FETCH_FOR_USER` here? To avoid jumping when the the data is reduced